compose/snippets/src/main/java/com/example/compose/snippets/text/TextSnippets.kt

```diff
@@ -839,3 +840,55 @@ private val firaSansFamily = FontFamily()
 
 val LightBlue = Color(0xFF0066FF)
 val Purple = Color(0xFF800080)
+
+// [START android_compose_text_auto_format_phone_number_validatetext]
+@Composable
+fun ValidateInput() {
+    class EmailViewModel : ViewModel() {
+        var email by mutableStateOf("")
+            private set
+
+        val emailHasErrors by derivedStateOf {
+            if (email.isNotEmpty()) {
+                // Email is considered erroneous until it completely matches EMAIL_ADDRESS.
+                !android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()
+            } else {
+                false
+            }
+        }
+
+        fun updateEmail(input: String) {
+            email = input
+        }
+    }
+
+    @Composable
+    fun ValidatingInputTextField(
+        email: String,
+        updateState: (String) -> Unit,
+        validatorHasErrors: Boolean
+    ) {
+        val emailViewModel = EmailViewModel()
```

Comment:
This should use the recommended pattern here: https://developer.android.com/develop/ui/compose/libraries#viewmodel

i.e. import ` implementation 'androidx.lifecycle:lifecycle-viewmodel-compose:2.8.5'`
and then `viewModel: MyViewModel = viewModel()`

Reply:
Doesn't work when I do that. If I try it in the function signature, I get an "expecting an element" error. (There's nothing to pass, which is why we're declaring the variable locally.)

But if I try declaring it outside the signature, as a local variable, as you describe:

val emailViewModel: EmailViewModel=emailViewModel()

I get an error that emailViewModel is undeclared.

Reply:
fixed it for you

Reply:
ty!